code/src/WijmoProvider/Features/CalculatedField.ts

```diff
@@ -0,0 +1,95 @@
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+namespace WijmoProvider.Feature {
+    function Evaluate(formula: OSFramework.OSStructure.Formula) {
+        let fn: OSFramework.OSStructure.Functions = formula.function;
+
+        let parsedValues = formula.values.map((val) => {
+            if (isNaN(parseInt(val))) return `$.${val}`;
+            return val;
+        });
+
+        switch (fn) {
+            case OSFramework.OSStructure.Functions.Avg:
+                let vals = parsedValues.join(' + ');
+                return `(${vals}) / ${parsedValues.length}`;
+            case OSFramework.OSStructure.Functions.Diff:
+                return parsedValues.join(' - ');
+            case OSFramework.OSStructure.Functions.Div:
+                return parsedValues.join(' / ');
+            case OSFramework.OSStructure.Functions.Max:
+                return `Math.max(${parsedValues.join(', ')})`;
+            case OSFramework.OSStructure.Functions.Min:
+                return `Math.min(${parsedValues.join(', ')})`;
+            case OSFramework.OSStructure.Functions.Mult:
+                return parsedValues.join(' * ');
+            case OSFramework.OSStructure.Functions.Sum:
+                return parsedValues.join(' + ');
+            default:
+                return '';
+        }
+    }
+
+    export class CalculatedField
+        implements
+            OSFramework.Feature.ICalculatedField,
+            OSFramework.Interface.IBuilder
+    {
+        private _grid: Grid.IGridWijmo;
+        private _calculatedFields: any;
+
+        constructor(grid: Grid.IGridWijmo) {
+            this._grid = grid;
+            this._calculatedFields = {};
+        }
+
+        private _validateValues(values, header) {
+            const isValid = values
+                .filter((val) => isNaN(parseInt(val)))
+                .every(
+                    (value) =>
+                        this._grid
+                            // get columns
+                            .getColumns()
+                            // filter number and currency only
+                            .filter(
+                                (col) =>
+                                    col.columnType ===
+                                        OSFramework.Enum.ColumnType.Currency ||
+                                    col.columnType ===
+                                        OSFramework.Enum.ColumnType.Number
+                            )
+                            // get binding
+                            .map((col) => col.config.binding)
+                            .indexOf(value) !== -1
+                );
+
+            if (!isValid) {
+                throw `The content of ${header} column is not being displayed because the given formula is not valid. Formula values must be bindings to Currency or Number columns or numeric values written as text.
```

Comment:
A good one to be converted to an error code in the future! (near future)

Reply:
Correct, will create a task in JIRA to implement error code logic in Data Grid Component